gateleen-routing/src/main/java/org/swisspush/gateleen/routing/RouterConfigurationParser.java

```diff
@@ -0,0 +1,113 @@
+package org.swisspush.gateleen.routing;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.oauth2.OAuth2FlowType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.swisspush.gateleen.core.util.StringUtils;
+import org.swisspush.gateleen.routing.auth.OAuthConfiguration;
+
+import javax.annotation.Nullable;
+import java.util.*;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class RouterConfigurationParser {
+
+    private static final Logger log = LoggerFactory.getLogger(RouterConfigurationParser.class);
+
+    private static final String REQUEST_HOPS_LIMIT_PROPERTY = "request.hops.limit";
+    private static final String AUTHCONFIGS_PROPERTY = "authConfigs";
+    private static final String FLOWTYPE_PROPERTY = "flowType";
+    private static final String CLIENTID_PROPERTY = "clientId";
+    private static final String CLIENTSECRET_PROPERTY = "clientSecret";
+    private static final String SITE_PROPERTY = "site";
+    private static final String TOKENPATH_PROPERTY = "tokenPath";
+    private static final String AUTHPATH_PROPERTY = "authPath";
+    private static final String SCOPES_PROPERTY = "scopes";
+    private static final String SUPPORTEDGRANTTYPES_PROPERTY = "supportedGrantTypes";
```

Comment:
Trivial, we could add some extra _, eg.

 SUPPORTED_GRANTTYPES_PROPERTY 
AUTH_CONFIGS_PROPERTY